artiq/dashboard/experiments.py

```diff
@@ -703,6 +703,26 @@ def submit(self, expurl):
             scheduling["priority"], scheduling["due_date"],
             scheduling["flush"]))
 
+    def submit_by_content(self, content, class_name, description=None):
```

Comment:
The experiment manager is dealing with the experiment GUI docks, the repos, etc.
It is not the place to define unrelated internal APIs.
Pass the scheduler connection to moninj instead and move that function there. ``_submit_task`` is a 2-line function that is not interesting to share.